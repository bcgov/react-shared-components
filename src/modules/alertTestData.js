import React from "react";
import { Alert } from "../components/alert/Alert";

export function generateAlert(icon, type, styling, element) {
  return <Alert icon={icon} type={type} styling={styling} element={element} />;
}
