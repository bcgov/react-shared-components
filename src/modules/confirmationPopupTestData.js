/* eslint-disable react/jsx-one-expression-per-line */
import React from "react";

const body = () => (
  <>
    <p>Your files will not be submitted.</p>
    <p>
      You will be returned to:
      <br />
      <b>Original</b> website
    </p>
  </>
);

export function getConfirmationPopupTestData() {
  return { body };
}
