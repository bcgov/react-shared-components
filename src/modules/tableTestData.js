const elements = [
  { name: "BCeID", value: "bobross42" },
  {
    name: "Last Name",
    value: "Ross",
  },
  { name: "First Name", value: "Bob" },
  { name: "Email Address", value: "bob.ross@example.com" },
];

export function getTableElementsTestData() {
  return elements;
}
