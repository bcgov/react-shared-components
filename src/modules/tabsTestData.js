import React from "react";

export const tabs = [
  {
    title: "Tab1",
    content: <p> Tab1 Content... </p>,
  },
  {
    title: "Tab2",
    content: <p> Tab2 Content... </p>,
  },
  {
    title: "Tab3",
    content: <p> Tab3 Content... </p>,
    isDisabled: true,
  },
];
