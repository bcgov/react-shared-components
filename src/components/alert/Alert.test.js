import React from "react";
import { MdError, MdCancel, MdCheckBox, MdInfo } from "react-icons/md";

import testBasicSnapshot from "../../TestHelper";
import { generateAlert } from "../../modules/alertTestData";

describe("Alert Component", () => {
  test("Matches the 'success' snapshot", () => {
    testBasicSnapshot(
      generateAlert(
        <MdCheckBox size={32} />,
        "success",
        "success-background",
        "This is a success message!"
      )
    );
  });

  test("Matches the 'warning' snapshot", () => {
    testBasicSnapshot(
      generateAlert(
        <MdError size={32} />,
        "warning",
        "warning-background",
        "This is a warning message!"
      )
    );
  });

  test("Matches the 'error' snapshot", () => {
    testBasicSnapshot(
      generateAlert(
        <MdCancel size={32} />,
        "error",
        "error-background",
        "This is an error message!"
      )
    );
  });

  test("Matches the 'info' snapshot", () => {
    testBasicSnapshot(
      generateAlert(
        <MdInfo size={32} />,
        "info",
        "info-background",
        "This is an info message!"
      )
    );
  });
});
