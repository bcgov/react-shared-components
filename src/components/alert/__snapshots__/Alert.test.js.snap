// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert Component Matches the 'error' snapshot 1`] = `
<div
  className="bcgov-display-box bcgov-error-background"
>
  <div
    className="bcgov-display-icon"
  >
    <div
      style={
        Object {
          "color": "#D8292F",
        }
      }
    >
      <svg
        fill="currentColor"
        height={32}
        size={32}
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width={32}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm5 13.59L15.59 17 12 13.41 8.41 17 7 15.59 10.59 12 7 8.41 8.41 7 12 10.59 15.59 7 17 8.41 13.41 12 17 15.59z"
        />
      </svg>
    </div>
  </div>
  <div
    className="bcgov-display-right-element"
  >
    This is an error message!
  </div>
</div>
`;

exports[`Alert Component Matches the 'info' snapshot 1`] = `
<div
  className="bcgov-display-box bcgov-info-background"
>
  <div
    className="bcgov-display-icon"
  >
    <svg
      fill="currentColor"
      height={32}
      size={32}
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width={32}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
      />
    </svg>
  </div>
  <div
    className="bcgov-display-right-element"
  >
    This is an info message!
  </div>
</div>
`;

exports[`Alert Component Matches the 'success' snapshot 1`] = `
<div
  className="bcgov-display-box bcgov-success-background"
>
  <div
    className="bcgov-display-icon"
  >
    <div
      style={
        Object {
          "color": "#2E8540",
        }
      }
    >
      <svg
        fill="currentColor"
        height={32}
        size={32}
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width={32}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
        />
      </svg>
    </div>
  </div>
  <div
    className="bcgov-display-right-element"
  >
    This is a success message!
  </div>
</div>
`;

exports[`Alert Component Matches the 'warning' snapshot 1`] = `
<div
  className="bcgov-display-box bcgov-warning-background"
>
  <div
    className="bcgov-display-icon"
  >
    <div
      style={
        Object {
          "color": "rgb(252, 186, 25)",
        }
      }
    >
      <svg
        fill="currentColor"
        height={32}
        size={32}
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width={32}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-2h2v2zm0-4h-2V7h2v6z"
        />
      </svg>
    </div>
  </div>
  <div
    className="bcgov-display-right-element"
  >
    This is a warning message!
  </div>
</div>
`;
